pyserini/index/pyutils.py

```diff
@@ -35,13 +35,105 @@ class IndexUtils:
     '''
 
     def __init__(self, index_dir):
-        self.object = JIndexUtils(JString(index_dir))
+        self.object = JIndexReaderUtils()
+        self.reader = self.object.getReader(JString(index_dir))
 
     class DocumentVectorWeight:
         NONE = JDocumentVectorWeight.NONE
         TF_IDF = JDocumentVectorWeight.TF_IDF
 
-    def dump_document_vectors(self, reqDocidsPath: str, weight: DocumentVectorWeight):
+    class Posting:
+        '''
+        Basic Posting class for Postings List
+        '''
+        def __init__(self, docid, term_freq, positions):
+            self.docid = docid
+            self.term_freq = term_freq
+            self.positions = positions
+
+        def __repr__(self):
+            repr = '(' + str(self.docid) + ', ' + str(self.term_freq) + ')'
+            if self.positions:
+                repr += ' [' + ','.join([str(p) for p in self.positions]) + ']'
+            return repr
+
+    def analyze_term(self, term):
+        '''
+        Parameters
+        ----------
+        term : str
+            Search term
+        Returns
+        -------
+        result : str
+            Stemmed term
+        '''
+        return self.object.analyzeTerm(self.reader, JString(term))
+
+    def get_term_counts(self, term):
+        '''
+        Parameters
+        ----------
+        term : str
+            Search term
+        Returns
+        -------
+        result : long, long
+            Collection frequency and document frequency of term
+        '''
+        term_map = self.object.getTermCounts(self.reader, JString(term))
+        return term_map.get(JString('collectionFreq')), term_map.get(JString('docFreq'))
+
+    def get_postings_list(self, term):
+        '''
+        Parameters
+        ----------
+        term : str
+            Search term
+        Returns
+        -------
+        result : list<Posting>
+            Postings list for term
+        '''
+        postings_list = self.object.getPostingsList(self.reader, JString(term))
+        result = []
+        for posting in postings_list.toArray():
+            result.append(self.Posting(posting.getDocid(), posting.getTF(), posting.getPositions()))
+        return result
+
+    def get_document_vector(self, docid):
+        '''
+        Parameters
+        ----------
+        docid : str
+            Lucene document ID
```

Comment:
I believe this is the docid as defined in the collection? call it "collection docid" to be unambiguous. This is different from the Lucene internal docid.

So "Lucene document ID" is ambiguous.
Also below.